### PR TITLE
PHP 8 support

### LINKED ISSUE #10: PHP 8 support
Is there anything preventing a change from `"php": "^5.6|^7.0"` to `"php": "^5.6|^7.0|^8.0"`? If there are other changes required, happy to submit a PR if you let me know what needs to be done.

### PR DESCRIPTION
**Pull request by ndench**
After creating #10, I figured the best way to check what issues exist for PHP 8 would be to run CI.

Closes: #10

### CODE REVIEW

**Comment by ndench** [comment]
Looks like we need to upgrade to phpunit 8.5 to get PHP 8.0 support. Also the PHP 7.3 and 7.4 builds appear to be using xdebug 3?

**Comment by AlaaSarhan** [comment]
Thanks @ndench for this PR. Are you planning to come back and look at the issues? I may also be able to put sometime on it this week to find, fix and release

**Comment by AlaaSarhan** [comment]
Closes #10 #11

**Comment by ndench** [comment]
@AlaaSarhan I'm happy to fix up issues, I just wanted to get your input first. Because it appears that we may have to drop support for PHP 5.6 and I thought there might be a bit of work in upgrading PHPUnit. Your PR looks pretty good though. I'll leave some comments on it.

**Comment by ndench** [comment]
@AlaaSarhan after pulling in the changes you made in #12 I just had increase the minimum PHP version to 7.1 to make everything play nicely. Hope this is ok.

File: composer.json [hunk 1]
```diff
@@ -3,11 +3,11 @@
     "description": "Docusign OAuth2 provider for League OAuth2 Client",
     "keywords": ["docusign", "oauth", "oauth2"],
     "require": {
-        "php": "^5.6|^7.0",
+        "php": "^7.1|^8.0",
         "league/oauth2-client": "^2.0"
     },
     "require-dev": {
-        "phpunit/phpunit": "^5.7|^6.0",
+        "phpunit/phpunit": "^5.7|^6.0|^9.0",
```
Comment (AlaaSarhan) [inline composer.json:10]:
```suggestion
        "phpunit/phpunit": "^7.0|^8.0|^9.0",
```

**Review by AlaaSarhan — APPROVED** [review]
This is looking great @ndench 🚀 Thanks so much for working on it. I will merge and prepare a major release right after

**Pull request merged**